### DISCUSSION TITLE
Feature request : Single click file folder opening

### DISCUSSION THREAD

**Discussion by samirkatyar** [kimlimjustin/xplorer#194] [General]
I'm using a linux and windows both.
in both the os im using a single click file and folder opening.

so i think many of us using a single click for opening a file and folder.

**Comment by lg-nineoneninethreenine** [comment]
I don't understand, are you asking for a single-click to open folders? I think that'd bring some trouble, because I think half of the community uses double click for opening and single click for selecting the files. Maybe can we integrate it as an optional feature @kimlimjustin?

**Reply by uahnbu** [reply]
I believe the default option is that every folder in the ***homescreen*** (where favorites, drives and recent files are listed), and this screen only, of all file exploring app I've used is accessible through a single click. To me it's a minor change that Xplorer use double click instead, as I don't find it hard to reform my muscle memory :joy:

**Comment by lg-nineoneninethreenine** [comment]
But I seem to have double click for them. I say that this can be implemented with a toggle, that'd be great, since if I switch onto another computer, I might install Xplorer, and I could use double-click to open and single-click to select.Whereas, you can enable the toggle and use single-click to open files and folders.All I recommend is a toggle, since it can go further to single-clickers.

**Comment by kimlimjustin** [comment]
Do you mean the ability to open the directory in only the _Home_ page, or also on opening folder and files too (which means you won't lose the ability to select a file by single clicking it tho)

**Reply by kimlimjustin** [reply]
Or maybe just add both options 🤔?

**Reply by lg-nineoneninethreenine** [reply]
Why not add a toggle too?

**Reply by kimlimjustin** [reply]
> Why not add a toggle too?

Do you mean the option to toggle it on the preference page on settings? that's what I was talking bout

**Reply by lg-nineoneninethreenine** [reply]
Yeah, that's what I meant too.

**Comment by uahnbu** [comment]
Idk but toggles only allow user to select multiple files without pressing `Ctrl/Cmd`, rite? Actually in Windows there's an option to make files appear as links, which allow single clicking. Then single click behavior is replaced by 1000ms-hovering.

**Reply by lg-nineoneninethreenine** [reply]
Yeah I know that, we can use this same feature for Xplorer. Trouble is, if you hover, and if you mouse gets a delay(happens rarely, but it might) or you mouseslipped, you'd select a file. That'd be kind of trouble. But I am looking forward to see this on the codebase.

**Reply by kimlimjustin** [reply]
I think it'll just something in the `Preference` page inside settings

**Reply by lg-nineoneninethreenine** [reply]
Exactly my point!

**Comment by kimlimjustin** [comment]
Update: implemented this preference option, please give it a try by building from source if you want to, otherwise, stay tuned for next release (prob end of this month)

**Reply by lg-nineoneninethreenine** [reply]
After I switch OSes, first step: get requirements and then, build Xplorer from source. Need to try this out, since if it works for my soon-to-be Zorin system, it will work on Gitpod. Let the PR be a draft, or I will close it. I hope to switch to Zorin during a two-day break, so after that, I will test this, fix Gitpod and get ready for the next release.
@kimlimjustin can you just say to Gitpod users that due to the rebrand, Xplorer isn't compatible with Gitpod as of now, once this is fixed, the text will be removed. Once it's fixed, I will remake the PR from Ubuntu POV, fix this whole thing. Hoping to do the install before release, if I don't, leave it, I will do this tho.

**Reply by kimlimjustin** [reply]
@im-coder-lg sure

**Reply by lg-nineoneninethreenine** [reply]
I am wishing to close the PR and reproduce+fix this locally as soon as I install Zorin(ubuntu-based) so you can close the PR, and change the fix part to `A fix is not totally yet in progress, any help would be appreciated` so that users can know that a fix is not presented yet.

**Reply by kimlimjustin** [reply]
I think no need again, I've made the dockerfile thing work, but it seems to be laggy with the `gitpod.yml` file, here's the working dockerfile on my debug account: https://github.com/kimlimjustin-debug/xplorer/blob/master/.gitpod.Dockerfile

**Reply by kimlimjustin** [reply]
I think I'll debug it manually on my own as it's almost done (only there's a bug on linux and I'll debug that in my VM), probs can be done today.
